File: Solutions/0110- Balanced Binary Tree/Solution.java
class Solution {
    public boolean isBalanced(TreeNode root) {
        return height(root) >= 0;
    }

    private int height(TreeNode root) {
        if (root == null)
            return 0;

        int left_height = height(root.left);
        int right_height = height(root.right);

        if (Math.abs(left_height - right_height) > 1 || left_height == -2 || right_height == -2)
            return -2;

        return Math.max(left_height,right_height) + 1;
    }


}

File: Solutions/0110- Balanced Binary Tree/Solution.py
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:

        def height(root):
            if not root:
                return 0

            right = height(root.right)
            left = height(root.left)

            if abs(left - right) > 1 or left == -2 or right == -2:
                return -2

            return max(right, left) + 1

        return height(root) != -2